contracts/ScoutToken.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.26;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Pausable.sol";
+import "@openzeppelin/contracts/access/extensions/AccessControlEnumerable.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+contract ScoutToken is Context, AccessControlEnumerable, ERC20Pausable {
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    bytes32 public constant VESTING_ROLE = keccak256("VESTING_ROLE");
+
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {
+        _grantRole(DEFAULT_ADMIN_ROLE, _msgSender());
+
+        _grantRole(MINTER_ROLE, _msgSender());
+        _grantRole(PAUSER_ROLE, _msgSender());
+    }
+
+    function mint(address to, uint256 amount) public virtual {
+        require(
+            hasRole(MINTER_ROLE, _msgSender()),
+            "Must have minter role to mint"
+        );
+        _mint(to, amount);
+    }
+
+    function mintForVesting(address to, uint256 amount) public virtual {
+        require(
+            hasRole(VESTING_ROLE, _msgSender()),
+            "Must have vesting role to mint for vesting"
+        );
+        _mint(to, amount);
+    }
+
+    function pause() public virtual {
+        require(
+            hasRole(PAUSER_ROLE, _msgSender()),
+            "Must have pauser role to pause"
+        );
+        _pause();
+    }
+
+    function unpause() public virtual {
```

Comment:
I see alot of use of virtuals here, which mainly applies for overriding from inheriting contracts.

What is the use case for having virtuals everywhere?